source/code/plugins/in_vminsights.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+
+class Fluent::VMInsights < Fluent::Input
+    require_relative 'oms_common'
+
+    require_relative 'VMInsightsDataCollector.rb'
+    require_relative 'VMInsightsEngine.rb'
+
+    Fluent::Plugin.register_input('vminsights', self)
+
+    config_param :tag, :string
+    config_param :poll_interval, :integer, :default => 60
+
+    def initialize
+        super
+        @instance_id = self.class.name + "(" + Time.now.to_s + ")"
+    end
+
+    def configure(conf)
+        super
+
+        begin
+            @heartbeat_uploader = conf[:MockMetricsEngine] || ::VMInsights::MetricsEngine.new
+            @heartbeat_upload_configuration = make_heartbeat_configuration
+        rescue Fluent::ConfigError
+            raise
+        rescue => ex
+            @heartbeat_upload_configuration = nil
```

Comment:
how are you e2e testing this?

Reply:
please send us info about that as we would release the shell bundle part of the change

Reply:
At this time, we have

- manually added the plugin to sample systems and restarted the omsagent,
- verified the data flowing through ODS, etc. into the workspace's Kusto tables is as expected.

Once we can get this code into DSC, we will submit the PR to connect the shell bundle version with the intelligence pack in test and verify again before going to production.